File: Liskovs/compliant/ShapeFactory.java

public class ShapeFactory {
    public Shape createShape() {
        // The post condition is that it always returns a shape
        return new Shape();
    }
    public void draw () {
        System.out.println("Drawing");
    }
}

class CircleFactory{
    public Circle createShape() {
        return new Circle();
    }
}

File: Liskovs/compliant/ShapeFactory.ts
class Shape {
    draw(): void {
        console.log("Drawing Shape");
    }
}

class Circle extends Shape {
    draw(): void {
        console.log("Drawing Circle");
    }
}

class ShapeFactory {
    createShape(): Shape {
        return new Shape();
    }

    draw(): void {
        console.log("Drawing");
    }
}

class CircleFactory extends ShapeFactory {
    createShape(): Circle {
        return new Circle();
    }
}
